linux/net/rina/serdes.c

```diff
@@ -791,8 +791,6 @@ static struct pdu_ser * pdu_serialize_gfp(gfp_t                       flags,
             ASSERT(dup_chksum_is_ok(tmp));
         }
 
-        pdu_destroy(pdu);
-
```

Comment:
Yeap, and this is wrong. Writing to the N-1 port can return -EAGAIN (try later, congestion now), and then the PDU has to be requeued. If the PDU has already been destroyed by pdu_serialize, you can't requeue it.